xmtp_mls/src/groups/sync.rs

```diff
@@ -219,7 +219,8 @@ impl MlsGroup {
                     openmls_group.merge_pending_commit(&provider)
                 {
                     log::error!("error merging commit: {}", err);
-                    openmls_group.clear_pending_commit();
+                    let _ = openmls_group.clear_pending_commit(provider.storage());
```

Comment:
how bad would it be if we error here and don't clear the commit? Previously `clear_pending_commit` wouldn't interact with storage, but now it does a write

Reply:
If the write fails somehow we wouldn't know and the pending commit wouldn't be cleared, I think we want to bubble the error with `?` here?

Reply:
Hmmmm... I'm not sure maybe @richardhuaaa might know?